olp-cpp-sdk-core/include/olp/core/client/ApiLookupClient.h

```diff
@@ -67,6 +73,46 @@ class CORE_API ApiLookupClient final {
   /// Alias for the parameters and responses.
   using LookupApiResponse = ApiResponse<OlpClient, ApiError>;
   using LookupApiCallback = std::function<void(LookupApiResponse)>;
+
+  ApiLookupClient(const HRN& catalog, const OlpClientSettings& settings);
+  ~ApiLookupClient();
+
+  /** @brief Gets a API for a single service sync, internally uses
+   * ApiLookupSettings from OlpClientSettings.
+   *
+   * @param service The service name.
+   * @param service_version Version of the required service.
+   * @param options The fetch option that you can use to set the source from
+   * which data should be fetched.
+   * @param context The `CancellationContext` instance that is used to cancel
+   * the request.
+   *
+   * @return `LookupApiResponse` that contains the OlpClient instance
+   * or an error.
+   */
+  LookupApiResponse LookupApi(const std::string& service,
+                              const std::string& service_version,
+                              FetchOptions options,
+                              CancellationContext context);
+
+  /** @brief Gets a API for a single service async, internally uses
+   * ApiLookupSettings from OlpClientSettings.
+   *
+   * @param service The service name.
+   * @param service_version Version of the required service.
+   * @param options The fetch option that you can use to set the source from
+   * which data should be fetched.
+   * @param callback The function callback used to receive the
+   * `LookupApiResponse` instance.
+   *
+   * @return The method used to call or to cancel the request.
+   */
+  CancellationToken LookupApi(const std::string& service,
+                              const std::string& service_version,
+                              FetchOptions options, LookupApiCallback callback);
+
+ private:
+  std::unique_ptr<ApiLookupClientImpl> impl_;
```

Comment:
Use std::shared_ptr, or how else would you be able to share this among repositories?